include/etl/span.h

```diff
@@ -938,6 +939,28 @@ namespace etl
     }
   };
 #endif
+
+  //*************************************************************************
+  /// Obtains a view to the object representation of the elements of the span s.
+  //*************************************************************************
+
+#if ETL_USING_CPP11
```

Comment:
Are you sure this is only implementable in C++11?
You could make it compatible with C++03 by replacing `constexpr` with `ETL_CONSTANT`, and `static_assert` with `ETL_STATIC_ASSERT`.

Reply:
Yes, `auto` is not available to C++03, you'll have to spell out the return type in full.
A great many embedded programmers are still updating legacy devices/development platforms that do not use any standard later than C++03. I try to allow C++03 compatibility wherever it is technically possible.

Reply:
I was a C programmer for about 12 years until somewhen around 2000, when I moved jobs and had the opportunity to look at C++. I was amazed to realise that I'd been implementing a lot of C++ functionality, in C, for years!